### ISSUE TITLE
tablename is incorrect for queries in brackets

### ISSUE THREAD

**Issue by abraxxa** [jplindstrom/p5-Test-DBIC-ExpectedQueries#6]
Those get generated by https://metacpan.org/pod/DBIx::Class::Helper::ResultSet::OneRow on Oracle:

```
#   Failed test 'Expected queries for tables'
    #   at xt/05forms.t line 80.
    # 
    # * Table: SELECT
    # Expected '0' selects for table 'SELECT', got '6'
    # Actually executed SQL queries on table 'SELECT':
    # SQL: (SELECT me.id, me.req_id, me.version, me.name FROM (
    #   SELECT me.id, me.req_id, me.version, me.name FROM tablename me WHERE ( req_id = ? )  ORDER BY version DESC
    # ) me WHERE ROWNUM <= ?
    # : '82', '1')
```

**Comment by jplindstrom** [comment]
Cheers!

I'm not sure what the desired behaviour here could be. Maybe it's to just ignore the first SELECT and try for the next one. Maybe it's to disregard all selects on unknown tables. Or just document it as it  currently is in the caveats section.

(I won't be writing a SQL parser for this :) )

**Comment by abraxxa** [comment]
Thanks for the quick response!
The desired behavior is to report this as a `select` for the table `tablename`.

**Comment by jplindstrom** [comment]
What I'm uncertain of is whether "if the table looks like "SELECT, look for the next thing that's FROM something" is going to cause weird outcomes for other SQL queries.

I might need to make the Caveats section more prominent, or maybe put this behind an option to avoid breaking backwards compatibility.

**Comment by abraxxa** [comment]
I looked at the code and it seems the `optional quote` regex (https://github.com/jplindstrom/p5-Test-DBIC-ExpectedQueries/blob/master/source/lib/Test/DBIC/ExpectedQueries/Query.pm#L19) captures the opening bracket so SELECT ends up being captured by the `capture table` regex.

I'd say in the case of a nested query the innermost table name should be reported.

**Comment by jplindstrom** [comment]
I just released 2.002, can you please test using that version (and the new option) and re-open this if it doesn't solve your problems?

https://metacpan.org/release/JOHANL/Test-DBIC-ExpectedQueries-2.002

**Comment by abraxxa** [comment]
Works for my current use-case.
Thanks for the quick fix and release!
If I find some other edge cases I'll open a new issue.
